api/v1alpha1/nodeobservability_types.go

```diff
@@ -23,21 +23,40 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+// +kubebuilder:validation:Enum=crio;kubelet;ebpf;custom
+type NodeObservabilityType string
+
+const (
+	CrioNodeObservabilityType    NodeObservabilityType = "crio"
+	KubeletNodeObservabilityType NodeObservabilityType = "kubelet"
+	EbpfNodeObservabilityType    NodeObservabilityType = "ebpf"
+	CustomNodeObservabilityType  NodeObservabilityType = "custom"
```

Comment:
I still would keep them use case based, like "CRI-O and kubelet CPU/Memory Profile collection". Let's also remove ebpf and custom for now since they do not work at all.